### PR TITLE
Add JFROG_ACCESS_TOKEN as a optional environemnt variable

### PR DESCRIPTION
**Pull request by alexeiser**
Adds JFROG_ACCESS_TOKEN environment variable per https://github.com/jfrog/terraform-provider-artifactory/issues/414

I didn't see a useful way to add a test case for this change, and one didn't exist on https://github.com/jfrog/terraform-provider-xray

### CODE REVIEW

**Comment by alexeiser** [comment]
have read the CLA Document and I hereby sign the CLA

**Comment by alexeiser** [comment]
recheck

File: README.md [hunk 1]
```diff
@@ -70,22 +70,22 @@ Requirements:
...
-With the script you can start one or two Artifactory instances using docker compose.   
-The license is not supplied, but a [30 day trial license can be freely obtained](https://jfrog.com/start-free/#hosted) and will allow local development. Make sure the license saved as a multi line text file. 
+With the script you can start one or two Artifactory instances using docker compose.
+The license is not supplied, but a [30 day trial license can be freely obtained](https://jfrog.com/start-free/#hosted) and will allow local development. Make sure the license saved as a multi line text file.
 
-Currently, acceptance tests **require access key** and don't support basic authentication and API key. To generate access key, please refer to the [official documentation](https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-GeneratingAdminTokens) 
+Currently, acceptance tests **require access key** and don't support basic authentication and API key. To generate access key, please refer to the [official documentation](https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-GeneratingAdminTokens)
```
Comment (chb0github) [inline README.md:79]: I'm not sure of the change here but I do see a grammar mistake that was always there.

...require an access key...

File: docs/index.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ The Artifactory provider supports two ways of authentication. The following meth
 
 ### Access Token
 Artifactory access tokens may be used via the Authorization header by providing the `access_token` field to the provider
-block. Getting this value from the environment is supported with the `ARTIFACTORY_ACCESS_TOKEN` variable
+block. Getting this value from the environment is supported with `JFROG_ACCESS_TOKEN` or `ARTIFACTORY_ACCESS_TOKEN` variables
```
Comment (chb0github) [inline docs/index.md:66]: I feel like it's important to mention what the intent is. I'm kind of directing this to jfrog team

File: docs/index.md [hunk 3]
```diff
@@ -92,7 +92,7 @@ provider "artifactory" {
 The following arguments are supported:
 
 * `url` - (Optional) URL of Artifactory. This can also be sourced from the `ARTIFACTORY_URL` environment variable.
-* `access_token` - (Optional) This can also be sourced from the `ARTIFACTORY_ACCESS_TOKEN` environment variable.
+* `access_token` - (Optional) This can also be sourced from `JFROG_ACCESS_TOKEN` or `ARTIFACTORY_ACCESS_TOKEN` environment variables.
 * `api_key` - (Optional) API key for api auth. Uses `X-JFrog-Art-Api` header.
```
Comment (chb0github) [inline docs/index.md:96]: This actually needs to be removed. It's no longer supported @danielmkn

**Review by chb0github — COMMENTED** [review]
Mostly grammar and doc changes. Things we should have caught before.

Thanks for your contribution @alexeiser

File: README.md (hunk 1 above)
Comment (alexeiser) [reply README.md]: My editor just removes trailing whitespace - I have updated the grammar.

**Comment by alexeiser** [comment]
No problem @chb0github - happy to help. I don't think I can address the other items, let me know if there is something more you would need from me on this PR.

File: docs/index.md (hunk 3 above)
Comment (alexhung) [reply docs/index.md:96]: Didn't we decided to keep `api_key` because we need it for some feature?

**Review by alexhung — APPROVED** [review]
LGTM. Just need to get confirmation on `api_key`

File: docs/index.md (hunk 3 above)
Comment (danielmkn) [reply docs/index.md:96]: yes, we decided to keep the `api_key`.

**Review by maheshjfrog — APPROVED** [review]

**Comment by danielmkn** [comment]
LGTM, but we need to add info to CHANGELOG.md

**Pull request merged**